client/chat/channel-info-card.tsx

```diff
@@ -0,0 +1,197 @@
+    )
+  } else if (channelInfo?.private && !isUserInChannel) {
+    subtitle = <ErrorText>This channel is private and requires an invite to join</ErrorText>
+  } else if (isUserBanned) {
+    subtitle = <ErrorText>You are banned from this channel</ErrorText>
+  } else if (channelInfo?.userCount) {
+    subtitle = (
+      <Body1>{`${channelInfo.userCount} member${channelInfo.userCount > 1 ? 's' : ''}`}</Body1>
+    )
+  }
+
+  let action
+  if (isUserInChannel) {
+    action = <RaisedButton label='View' onClick={onViewClick} />
+  } else if (channelInfo?.private || isUserBanned) {
+    action = <RaisedButton label='Join' disabled={true} />
+  } else if (channelInfo) {
+    action = <RaisedButton label='Join' disabled={isJoinInProgress} onClick={onJoinClick} />
+  }
+
+  return (
+    <Container>
+      {isLoading ? (
+        <StyledLoadingDotsArea />
+      ) : (
+        <>
+          {icon}
+          <ChannelInfoContainer>
+            <Headline6>{channelName}</Headline6>
+            {subtitle}
+          </ChannelInfoContainer>
+          {action}
+        </>
+      )}
+    </Container>
+  )
+}
+
+export interface ConnectedChannelInfoCardProps {
+  channelId: SbChannelId
+  channelName: string
+}
+
+export function ConnectedChannelInfoCard({
+  channelId,
+  channelName,
+}: ConnectedChannelInfoCardProps) {
+  const dispatch = useAppDispatch()
+  const channelInfo = useAppSelector(s => s.chat.idToInfo.get(channelId))
+  const isUserInChannel = useAppSelector(s => s.chat.joinedChannels.has(channelId))
+
+  const [findChannelError, setFindChannelError] = useState<Error>()
+  const [joinChannelError, setJoinChannelError] = useState<Error>()
+  const [isJoinInProgress, setIsJoinInProgress] = useState(false)
+
+  useEffect(() => {
+    const abortController = new AbortController()
+
+    dispatch(
+      getChannelInfo(channelId, {
+        signal: abortController.signal,
+        onSuccess: () => setFindChannelError(undefined),
+        onError: err => setFindChannelError(err),
+      }),
+    )
+
+    return () => {
+      abortController.abort()
+    }
+  }, [channelId, dispatch])
+
+  const onViewClick = useStableCallback(() => {
+    navigateToChannel(channelId, channelName)
+  })
+
+  const onJoinClick = useStableCallback(() => {
+    setIsJoinInProgress(true)
+    dispatch(
+      joinChannel(channelName, {
+        onSuccess: channel => navigateToChannel(channel.id, channel.name),
+        onError: err => {
+          setJoinChannelError(err)
+          setIsJoinInProgress(false)
+        },
+      }),
+    )
+  })
+
+  const isChannelNotFound =
+    isFetchError(findChannelError) && findChannelError.code === ChatServiceErrorCode.ChannelNotFound
+  const isUserBanned =
+    isFetchError(joinChannelError) && joinChannelError.code === ChatServiceErrorCode.UserBanned
+
+  return (
+    <ChannelInfoCard
+      channelName={channelName}
+      channelInfo={channelInfo}
+      isLoading={!channelInfo && !findChannelError}
```

Comment:
This is really not a good structure for this, and will be a source of many bugs and  a lot of confusion, just as it has been in the matchmaking reducer. These are not separate states, they are all just potential points in the graph of states, but you're passing them piecemeal. A user cannot be banned from a channel that does not exist, the user cannot be in a channel that does not exist, etc.

I'd still prefer that this kind of state is managed within the component itself (like I said before, replacing getChannelInfo/joinChannel in the test page is almost certainly a better solution that will result in more 'real' behavior being tested), but if you must pass it this way, you should be passing that state on a single prop, not 5 props